### PR TITLE
User list ui

### PR DESCRIPTION
**Pull request by kulmann**
This PR provides a very simple user listing UI. Implements https://github.com/owncloud/product/issues/96

Known limitations:
- There is no pagination yet (planned for the future)
- ~~Looks like the `accountEnabled` boolean is not maintained, yet (so accounts appear to be disabled in the UI)~~
- Avatars all look the same (seems to be a default image)

In order to get it running, you need to 
- have the proxy url as server in the ocis-web config
- add an endpoint `/api/v0/accounts` pointing to `http://localhost:9181` to the proxy config
- add an `external_apps` entry in the ocis-web config, named `accounts` and pointing to `http://localhost:9181/accounts.js`

It might be necessary to remove `/var/tmp/ocis-accounts/accounts` if you see less than 5 users in the list. Accounts only get recreated if the folder is missing.

## Screenshot of the current state
Screenshot 2020-07-01 at 15 42 19

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- pkg/server/http/option.go  2
- pkg/server/http/server.go  3
- pkg/assets/assets.go  6
- pkg/assets/option.go  2
- rollup.config.js  2
- pkg/metrics/metrics.go  1
- ui/store/index.js  3
- pkg/assets/embed.go  9
         

Clones added
============
- pkg/server/http/option.go  1
- pkg/assets/option.go  1
- ui/client/accounts/index.js  110
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084&bid=18939247)

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import vue from 'rollup-plugin-vue'
+import { terser } from 'rollup-plugin-terser'
```
Comment (ownclouders) [inline rollup.config.js:2]: Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/assets/embed.go  9
- pkg/server/http/option.go  2
- ui/store/index.js  3
- pkg/assets/option.go  2
- pkg/server/http/server.go  3
- pkg/assets/assets.go  6
- pkg/metrics/metrics.go  1
         

Clones added
============
- pkg/server/http/option.go  1
- pkg/assets/option.go  1
- ui/client/accounts/index.js  110
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084&bid=18939247)

**Comment by kulmann** [comment]
@refs I added you as a reviewer because you were the one setting up the extension initially and you worked on ocis-settings coming from the boilr template. This PR does some refactoring to align more with the boilr project style, hopefully making it easier for other developers to find their way through this extension, if they know e.g. ocis-hello already.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- pkg/server/http/option.go  2
- pkg/assets/assets.go  6
- rollup.config.js  2
- pkg/assets/option.go  2
- pkg/assets/embed.go  9
- pkg/server/http/server.go  3
- ui/store/index.js  3
- pkg/metrics/metrics.go  1
         

Clones added
============
- pkg/server/http/option.go  1
- pkg/assets/option.go  1
- ui/client/accounts/index.js  110
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084&bid=18939247)

File: rollup.config.js (hunk 1 above)
Comment (ownclouders) [inline rollup.config.js:2]: Codacy found an issue: [Unnecessary block.](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/assets/embed.go  9
- pkg/assets/assets.go  6
- pkg/server/http/server.go  3
- ui/store/index.js  3
- pkg/assets/option.go  2
- pkg/server/http/option.go  2
- pkg/metrics/metrics.go  1
         

Clones added
============
- ui/client/accounts/index.js  110
- pkg/assets/option.go  1
- pkg/server/http/option.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084&bid=18939247)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/metrics/metrics.go  1
- pkg/server/http/server.go  3
- pkg/assets/assets.go  6
- pkg/server/http/option.go  2
- ui/store/index.js  3
- pkg/assets/embed.go  9
- pkg/assets/option.go  2
         

Clones added
============
- pkg/server/http/option.go  1
- ui/client/accounts/index.js  110
- pkg/assets/option.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084&bid=18939247)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/server/http/server.go  3
- ui/store/index.js  3
- pkg/server/http/option.go  2
- pkg/assets/assets.go  6
- pkg/assets/option.go  2
- pkg/metrics/metrics.go  1
- pkg/assets/embed.go  9
         

Clones added
============
- ui/client/accounts/index.js  110
- pkg/server/http/option.go  1
- pkg/assets/option.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084&bid=18939247)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/server/http/option.go  2
- pkg/metrics/metrics.go  1
- pkg/assets/assets.go  6
- pkg/assets/embed.go  9
- pkg/assets/option.go  2
- ui/store/index.js  3
- pkg/server/http/server.go  3
         

Clones added
============
- pkg/server/http/option.go  1
- ui/client/accounts/index.js  110
- pkg/assets/option.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084&bid=18939247)

**Comment by kulmann** [comment]
CI is green. please review @refs @butonic @C0rby 
Ready for testing @individual-it

**Comment by kulmann** [comment]
Afaik @butonic wants to continue working with the accounts service. I would prefer getting this PR merged quickly and do UI tests in a separate PR @individual-it

File: pkg/proto/v0/accounts.proto [hunk 2]
```diff
@@ -18,13 +18,14 @@ service AccountsService {
     rpc ListAccounts(ListAccountsRequest) returns (ListAccountsResponse) {
         // List method maps to HTTP GET
         option (google.api.http) = {
-            get: "/v0/accounts"
+            post: "/api/v0/accounts/accounts-list",
```
Comment (butonic) [inline pkg/proto/v0/accounts.proto:21]: hmmmm .... no ... net really ,,,, urgh

File: pkg/flagset/flagset.go [hunk 3]
```diff
@@ -36,38 +36,60 @@ func RootWithConfig(cfg *config.Config) []cli.Flag {
...
-			Destination: &cfg.Server.Name,
+			Name:        "http-namespace",
+			Value:       "com.owncloud.web",
+			Usage:       "Set the base namespace for the http namespace",
+			EnvVars:     []string{"ACCOUNTS_HTTP_NAMESPACE"},
+			Destination: &cfg.HTTP.Namespace,
+		},
+		&cli.StringFlag{
+			Name:        "http-addr",
+			Value:       "localhost:9181",
```
Comment (butonic) [inline pkg/flagset/flagset.go:47]: bind to 0.0.0.0

File: pkg/flagset/flagset.go [hunk 4]
```diff
@@ -36,38 +36,60 @@ func RootWithConfig(cfg *config.Config) []cli.Flag {
...
-			Destination: &cfg.Server.Namespace,
+			Usage:       "Set the base namespace for the grpc namespace",
+			EnvVars:     []string{"ACCOUNTS_GRPC_NAMESPACE"},
+			Destination: &cfg.GRPC.Namespace,
 		},
 		&cli.StringFlag{
-			Name:        "address",
-			Aliases:     []string{"addr"},
+			Name:        "grpc-addr",
 			Value:       "localhost:9180",
```
Comment (butonic) [inline pkg/flagset/flagset.go:68]: bind to 0.0.0.0

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/server/http/server.go  3
- pkg/assets/assets.go  6
- ui/store/index.js  3
- pkg/assets/embed.go  9
- pkg/assets/option.go  2
- pkg/metrics/metrics.go  1
- pkg/server/http/option.go  2
         

Clones added
============
- ui/client/accounts/index.js  110
- pkg/assets/option.go  1
- pkg/server/http/option.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-accounts/pullRequest?prid=5785084&bid=18939247)

**Pull request merged**